tools/oeapkman/oeapkman

```diff
@@ -0,0 +1,396 @@
+#!/bin/bash
+# Copyright (c) Open Enclave SDK contributors.
+# Licensed under the MIT License.
+
+# URL to fetch alpine root fs from About 2.5 MB download.
+ALPINE_ROOTFS_URL="https://dl-cdn.alpinelinux.org/alpine/v3.14/\
+releases/x86_64/alpine-minirootfs-3.14.0-x86_64.tar.gz"
+
+# Use a recent version of proot. With older versions, apk would
+# report messages like "2 errors", that are innocuous, but catch
+# the eye.
+PROOT_URL="https://gitlab.com/proot/proot/-/jobs/981080842/\
+artifacts/download?file_type=archive"
+
+# Name of the folder where root filesystem will be extracted to.
+ALPINE_FS="alpine-fs"
+
+APORTS_BRANCH=3.13-stable
+
+# Figure out the user's home folder. If the command is executed
+# as sudo, then we use $SUDO_USER's home folder and not the
+# root's home folder.
+if [ -n "$SUDO_USER" ]; then
+    HOME_DIR=/home/${SUDO_USER}
+else
+    HOME_DIR=~
+fi
+
+OEAPKMAN_DIR=${HOME_DIR}/.oeapkman
+CURRENT_DIR=$(pwd)
+
+LOCK_FILE="/var/lock/oeapkman.lock"
+
+print_usage()
+{
+    usage="\
+oeapkman -  package manager and toolbox for enclave development.
+            version 0.1a, alpine-linux 3.14-stable
+
+oeapkman uses \`unshare\`/\`proot\` to manage an installation of alpine-linux.
+Alpine linux exists as a normal directory on the host machine and thus headers
+and libraries can be transparently used to build enclaves on the host machine.
+oeapkman also sets up the a development environment for fine-tuning and building
+packages specifically for enclaves.
+
+Links:
+Open Enclave SDK  : https://openenclave.io/sdk/
+Alpine Linux Wiki : https://wiki.alpinelinux.org/wiki/Main_Page
+Building Packages : https://wiki.alpinelinux.org/wiki/Abuild_and_Helpers
+Package Search    : https://pkgs.alpinelinux.org/packages
+Search Contents   : https://pkgs.alpinelinux.org/contents
+Proot             : https://proot-me.github.io/
+
+usage: oeapkman COMMAND [<ARGUMENTS>]
+
+Package installation and removal:
+  search     Search for packages matching given patterns.
+             E.g:
+                oeapkman search sqlite quickjs
+             Hint: Search for static and dev versions of libraries.
+             Hint: Use the \`Search Contents\` link above to find out what package
+                   contains the header/library files you are looking for.
+  add        Install packages
+             E.g:
+                oeapkman add sqlite-static sqlite-dev
+                oeapkman add quickjs
+  del        Remove given packages.
+
+Using packages:
+  root       The root folder of this alpine-linux instance. Header and library paths
+             can be specified relative to the root path.
+             E.g:
+             Specify header paths to compiler:
+                -I \`oeapkman root\`/usr/include/c++/10.2.1 -I \`oeapkman root\`/usr/include
+             Specify library paths to linker:
+                -L \`oeapkman root\`/usr/lib
+
+Execute commands:
+  exec       Execute a command in the current working directory. Current user's
```

Comment:
Similar to above, I'd prefer expanding this to "execute". (If you really just want to avoid typing, then you could do a strncmp match instead of a strcmp, to allow a prefix being typed, when parsing args, especially if you don't think you'll add other commands in the future that start with the same letter/prefix)

Reply:
See rationale above.